### DISCUSSION TITLE
Beginner question: setting up Git with Terminal on Mac

### DISCUSSION THREAD

**Discussion by Cellomaster87** [community/community#21880] [New to GitHub]
<p>Dear all,</p>
<p>this is my first ever post and, please, forgive me if this sounds as a silly question but I have browsed many guides and posts and not found any solution to this.</p>
<p>As a complete beginner with Git I am studying the ’ <strong>Pro Git</strong>’ book by <em>Scott Chacon</em> and <em>Ben Straub</em>.</p>
<p>The authors suggest doing everything via the Command Line (I am on macOS High Sierra 10.13.4) so I am following their instructions but … when I get to </p>
<p>“ <strong>Initializing a Repository in an Existing Directory</strong> ”</p>
<p>I get asked to go to the directory I want to initialise in Terminal and to write its path followed by ‘git init’. </p>
<p>When I do that on an Xcode project folder I get the error: ‘it is a directory’. Fine.</p>
<p>Then I simply write $ cd \Users&amp;lt;username&gt;&amp;lt;nameOfAFolder&gt; git init and press Return and I just get the title of the Terminal window changed to &lt;nameOfAFolder&gt; … while the book said I should get a ‘.git’ empty folder somewhere. </p>
<p>Could you tell me what am I making wrong? </p>
<p>I have already installed GitHub Desktop so I already have a GitHub folder into my Documents directory but I wanted to try the Command Line way anyway. </p>
<p>My doubt is also: is Git wanting a ‘home’ directory where all of my repositories will be stored or once I drag a folder to GitHub Desktop / manage to initialise it via Terminal it will be connected directly?</p>
<p>Thank you very much for yourtime and patience.</p>

**Comment by Jahhein** [comment]
<p>I’d like to try an revise this comment later when I have more time however here is as much as I can get out.</p>
<p>In whatever directory you are in, are you typing <code>ls -A</code> when you look for the .git folder in the terminal? Finder won’t show it by default, and you must specific with terminal commands as the -A means show all files.</p>
<p>dotfiles like .git are any file with a dot before the name, which aren’t shown in Finder by default, and must be specified to show with the <code>ls</code> command in the terminal.</p>

**Comment by Cellomaster87** [comment]
<p>Thank you Jahhein!</p>
<p>I have actually made some progress in the meantime.</p>
<p>Managed to show the hidden folders with the shortcut ‘Cmd-Shift-.’ and see all the mess that I had been doing! </p>
<p>Using the ‘cd’ command before the desired directory I managed to focus the Terminal on that very directory.</p>
<p>Then I ran ‘git init’ which created the .git folder inside.</p>
<p>Then ‘git add .’ to add all the files to the repository.</p>
<p>Then ‘git commit’ which prompted me to write a message, which I did in the last line after the lot of #### and file paths, but then it seems I am stuck. </p>
<p>How do I get out of there?</p>
<p>Thank you for your time!</p>

**Comment by Jahhein** [comment]
<p>Did you actually create any files that could be added to the commit?</p>
<p>Whatever folder/directory you are in, wether it be on a USB, inside a file in your Downloads folder, or wherever, whatever Folder/Directory you are in when you type <code>git init</code> is the top level of the repository.</p>
<p>Whats missing from your explaination is any file that could have been commited. type these commands out inside your repo folder and try again.</p>
<p>git init; echo “Hello World!” &gt;&gt; README.md ; git add . ; git commit -am “Test commit”;</p>
<p>copy the above series of terminal commands inside the folder you’ve been doing it, and then if it doesn’t work, do it in a new one somewhere.</p>

**Comment by Cellomaster87** [comment]
<p>I chose a folder of an Xcode project I am using to learn Swift.</p>
<p>So now there is a .git folder with all of the files inside. </p>
<p>When I type ‘git status’ it gives me a list of files in green color.</p>
<p>Just I could not get out of the Message part at the end when I entered ‘git commit’. </p>
<p>But maybe it was because I had not set the Editor properly. </p>
<p>It was set to Atom from within GitHub Desktop but it didn’t want to launch it from Terminal.</p>
<p>I changed it via the --global command to be Emacs and it worked … just it didn’t launch the app (and, from GitHub Desktop’s Preferences it seems Atom is still the preferred one without any chance to change it … maybe this is the issue?).</p>

**Comment by Jahhein** [comment, marked as answer]
<p>Then type the commit command like this:</p>
<p>git commit -am “here is the git commit message which bypasses git opening up an editor”</p>
<p>Sadly this message board doesn’t support code formating in the posts. anything between backticks `` is the command in previous posts of mine. Whats between the backticks is what the command is. don’t include them.</p>
<p>To change the git configuration for the editor the command is…</p>
<p>git config --global core.editor TheEditorYouWantHere</p>
<p>Just In case you changed it while inside the repo which takes presidence over global git config, run this while inside the repo directory:</p>
<p>git config --local core.editor TheEditorYouWantHere</p>
<p>personally i’d just use nano/terminal emacs in the terminal because you’re already in the terminal.</p>

**Comment by Cellomaster87** [comment]
<p>Thank you very much!</p>
<p>This solved it!</p>
<p>Thank you for guiding me through my first step in this new beautiful world!</p>

**Comment by Jahhein** [comment]
<p>You’re very welcome! Glad I could help you.</p>
<p>You will in time learn how it all works. Just realize all the commands you type are like mini programs. <code>cd</code> change directory, <code>ls</code> list, <code>pwd</code> print working directory, etc.</p>
<p>Git is just simply a program without an interface. Don’t be intimidated. Its all repitition and memorization. When in doubt always remember the <code>man</code> command for manual pages.</p>
<p>man git</p>
<p>man git-commit</p>
<p>man git-add</p>
<p>or…</p>
<p>git help commit</p>
